### ISSUE TITLE
feature request: add support for lolcat

### ISSUE THREAD

**Issue by ehaupt** [poetaman/arttime#42]
It would be nice to have support for a [lolcat](https://github.com/busyloop/lolcat) color filter :smiley_cat:

**Comment by poetaman** [comment]
@ehaupt Yes it would be a nice feature to have. It's actually pretty simple to achieve that, though I didn't add it as there are so many filters possible, and I wasn't sure about the best interface. Consider this problem: Filters tend to have a lot of options, and every user would want a slightly different option. So let's say we add an option ``--filter '<filter command_with_options>'``, if passed arttime will execute this string '<filter command_with_options>' each time a new text art is loaded in arttime. While on a home system this might not be an issue. But adding an option like ``--filter '<filter command_with_options>'`` could make something like arttime an eye sore for ones IT team IMO. What are your thoughts on it?

There are two other ways we can go about:
1. Add an option ``--filter <name>`` the string name won't be evaluated but act as a key to a set of filters we support with certain standard options. Like ``--filter lolcat`` would execute something like ``command lolcat -f -r``...
2. arttime implements a function in zsh to add random colors, and support both 16/256 colors.

Also, adding @romkatv if he would like to share any insights/experience on security from his zsh coding experience.

---

Arttime already has some files with color filter applied using lolcat for a simple disco effect. Try the following:

```
arttime --nolearn -a butterfly2 -b butterfly3
```

To immediately get that same effect on any other art, do something like this (example for text art``news``):
```
cd /path/to/share/arttime/textart/
lolcat -f -r news >news_lol1
lolcat -f -r news >news_lol2
arttime --nolearn -a news_lol1 -b news_lol2
```

**Comment by ehaupt** [comment]
Thank you for your explanation. Other project use an argument to pass arguments to external commands:

Eg.: `scp(1)`:

```
     -o ssh_option
             Can be used to pass options to ssh in the format used in
             ssh_config(5).  This is useful for specifying options for which
             there is no separate scp command-line flag.  For full details of
             the options listed below, and their possible values, see
             ssh_config(5).
```

But your proposed solution is perfectly reasonable. Please feel free to close the issue.

**Comment by poetaman** [comment]
Some update: It is recommended to run all pipes involving arttime in a subshell. I have updated the command in example above by wrapping it in ``()``.
